nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2019 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.util.Trees;
+import com.uber.nullaway.fixserialization.qual.AnnotationFactory;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.lang.model.element.Element;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/** Config class for Fix Serialization package. */
+public class FixSerializationConfig {
+
+  /**
+   * If activated, for all reporting errors, NullAway will serialize information and suggests type
+   * changes to resolve them, in case these errors could be fixed by adding a @Nullable annotation.
+   * These data will be written at output directory.
+   */
+  public final boolean suggestEnabled;
+  /**
+   * If activated, serialized information of a fix suggest will also include the enclosing method
+   * and class of the element involved in error.
+   */
+  public final boolean suggestEnclosing;
+  /** If activated, NullAway will write reporting errors in output directory. */
```

Comment:
Why do we need both `logErrorEnabled` and also `suggestEnabled`?

Reply:
In case where ```AutoFixer``` is performing ```Deep Search```, it needs to know both new errors and new fixes suggested in every run to create the ```fix chain tree```.

Reply:
Sorry, so is `logErrorEnabled` set to `false` when not doing deep search?

Reply:
Sure, I removed ```logErrorEnabled``` flag since in all runs of ```AutoFixer``` it is ```active```.

Below is the summary the commit: 9960615c0f7bd2e9bcfda644270649a942c99a39
1. Renamed ```FixSerializationConfig``` to ```SerializationConfig```.
2. Renamed ```Config.fixSerializationIsActive()``` to ```Config.serializationIsActive()```
3. Renamed error prone cli flag ```WriteFixMetadata``` to ```SerializeMetadata```.
4. Renamed error prone cli flag ```FixMetadataConfigPath``` to ```SerializationConfigPath```.
5. Error serialization is activated by default.
6. Updated all java docs to reflect the changes above.